processor/deltatocumulativeprocessor/internal/data/summ_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package data
+
+import (
+	"testing"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data/datatest"
+)
+
+// SummaryDP is a test helper for summary data points
+type SummaryDP struct {
+	Sum       float64
+	Count     uint64
+	Quantiles []QuantileValue
+}
+
+type QuantileValue struct {
+	Quantile float64
+	Value    float64
+}
+
+// Into converts SummaryDP to pmetric.SummaryDataPoint
+func (s SummaryDP) Into() pmetric.SummaryDataPoint {
+	dp := pmetric.NewSummaryDataPoint()
+	dp.SetSum(s.Sum)
+	dp.SetCount(s.Count)
+
+	// Only add quantiles if we have them
+	if len(s.Quantiles) > 0 {
+		quantiles := dp.QuantileValues()
+		for _, q := range s.Quantiles {
+			qv := quantiles.AppendEmpty()
+			qv.SetQuantile(q.Quantile)
+			qv.SetValue(q.Value)
+		}
+	}
+	return dp
+}
+
+// equalSummaryDataPoints compares two summary data points, treating nil and empty quantile slices as equivalent
+func equalSummaryDataPoints(expected, actual pmetric.SummaryDataPoint) bool {
```

Comment:
why is this custom logic required? what makes `datatest.Is.Equal` insufficient?

Reply:
I just removed that function. The idea was to compare quantiles only when they exist, otherwise they would be null.